### PR TITLE
[24.0] Replace busybox:ubuntu-14.04 image with busybox:1.36.1-glibc

### PR DESCRIPTION
**Pull request by mvdbeek**
We've (hopefully) reuploaded all docker images with deprecated manifests, but we have no control over this image.
I hope replacing this is as easy as this.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 1]
```diff
@@ -457,7 +457,7 @@
         <!-- Destinations define details about remote resources and how jobs
              should be executed on those remote resources.
          -->
-        <destination id="local" runner="local"/>
+          <destination id="local" runner="local" />
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]: Some of the indentation changes seem wrong to me, maybe use `xmllint` to reformat it?

Comment (mvdbeek) [reply lib/galaxy/config/sample/job_conf.xml.sample_advanced]: Not sure what vscode did there, plain old sed did the trick now.

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 2]
```diff
@@ -588,13 +588,13 @@
                trust tool's specified container - a destination wide override
                can be set. This will cause all jobs on this destination to use
                that docker image. -->
-          <!-- <param id="docker_container_id_override">busybox:ubuntu-14.04</param> -->
+          <!-- <param id="docker_container_id_override">busybox:1.36.1-glibc'</param> -->
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
          <!-- <param id="docker_container_id_override">busybox:1.36.1-glibc</param> -->
```

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 3]
```diff
@@ -588,13 +588,13 @@
...
 
           <!-- Likewise, if deployer wants to use docker for isolation and
                does trust tool's specified container - but also wants tool's not
                configured to run in a container the following option can provide
                a fallback. -->
-          <!-- <param id="docker_default_container_id">busybox:ubuntu-14.04</param> -->
+          <!-- <param id="docker_default_container_id">busybox:1.36.1-glibc'</param> -->
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
          <!-- <param id="docker_default_container_id">busybox:1.36.1-glibc</param> -->
```

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 4]
```diff
@@ -674,11 +674,11 @@
                the enabled container types for this destination will be used.
           -->
           <param id="container_override">
-              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:ubuntu-14.04</container>
+              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:1.36.1-glibc'</container>
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:1.36.1-glibc</container>
```

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 5]
```diff
@@ -674,11 +674,11 @@
...
               <container type="singularity" shell="/bin/sh" resolve_dependencies="false">/path/to/default/container</container>
           </param>
           <param id="container">
-              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:ubuntu-14.04</container>
+              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:1.36.1-glibc'</container>
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
              <container type="docker" shell="/bin/sh" resolve_dependencies="false">busybox:1.36.1-glibc</container>
```

File: lib/galaxy/config/sample/job_conf.xml.sample_advanced [hunk 6]
```diff
@@ -966,7 +966,7 @@
             container ID will be passed along to condor as
             the docker_image submission parameter.
             -->
-            <!-- <param id="docker_default_container_id">busybox:ubuntu-14.04</param> -->
+            <!-- <param id="docker_default_container_id">busybox:1.36.1-glibc'</param> -->
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/job_conf.xml.sample_advanced]:
```suggestion
            <!-- <param id="docker_default_container_id">busybox:1.36.1-glibc</param> -->
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**